nearly finished caseinsensitivedict

gwriterk opened the pull request:
merge get method and test to master branch

gwriterk committed:
yoooooo

gwriterk committed:
added get method and test

hopkiw commented on test_caseinsensitivedict.py:
```diff
@@ -67,6 +67,16 @@ def test_copy(self):
+        myinstance['Kaisa'] = 5
+        self.assertIn('Kaisa', myinstance)
+        self.assertNotIn('Jhin', myinstance)
+        value = myinstance.get('Kaisa')
```
add a case where we `.get()` the string 'kaisa' (all lower case). it should work for a case-insensitive dictionary and in this case it should return 5.

hopkiw commented on test_caseinsensitivedict.py:
```diff
@@ -100,6 +110,13 @@ def test_keys(self):
         # popitem()
         #     Remove and return a (key, value) pair from the dictionary.  Pairs
         #     are returned in LIFO order.
+    def test_popitem(self):
```
FYI these methods are not working, so shouldn't be merged to the master branch, until they're done.

hopkiw replied:
if these are working now, say so and resolve this comment

hopkiw commented on test_caseinsensitivedict.py:
```diff
@@ -122,7 +139,10 @@ def test_keys(self):
     def test_values(self):
```
this test method also not working yet

gwriterk committed:
fixed popitem method and added setdefault method. get and values methods still have issues

hopkiw commented on test_caseinsensitivedict.py:
```diff
@@ -67,10 +67,18 @@ def test_copy(self):
+    def test_get(self):
+        myinstance = CaseInsensitiveDict()
+        myinstance['Kaisa'] = 5
+        myinstance['kaisa'] = 5
```
what does this line do?

hopkiw commented on test_caseinsensitivedict.py:
```diff
@@ -67,10 +67,18 @@ def test_copy(self):
+        nonvalue = myinstance.get('Jhin')
+        self.assertEqual(value, 5)
+        self.assertEqual(nonvalue, None)
+        self.assertEqual(myinstance['kaisa'], 5)
```
using the `name[key]` syntax will invoke a specific method in our class, but *not* use the `get()` method. do you remember what method it will use?

the purpose of `test_get()` is to test the `get()` method, so change this here

hopkiw commented on test_caseinsensitivedict.py:
```diff
@@ -100,12 +108,27 @@ def test_keys(self):
+        self.assertIn('Kaisa', myinstance)
+        item = myinstance.popitem()
+        self.assertEqual(item, ('Kaisa', 5))
+        self.assertNotIn(item, myinstance)
```
what kinds of things can we use the 'in' operator on a dictionary with? what kind of thing are you using here?

hopkiw replied:
can we try adding 3 items and then popping all 3 (testing each value as you go), and confirming it is empty at the end?

hopkiw commented on test_caseinsensitivedict.py:
```diff
@@ -100,12 +108,27 @@ def test_keys(self):
+        myinstance['Kaisa'] = 5
+        item = myinstance.setdefault('Kaisa')
+        self.assertEqual(item, 5)
+        default_item = myinstance.setdefault('Jhin')
```
`setdefault()` can take one or two parameters, but you're only testing the single parameter case

gwriterk committed:
fixed methods

fixed __getitem__ method, __setitem__ method, values method, get method, and popitem method

gwriterk committed:
updated methods view

updated test document to reflect completed methods and tests
